tests/tc_billing_api.rb

```diff
@@ -0,0 +1,284 @@
+# Load the gem
+require 'pnap_billing_api'
+require 'oauth2'
+
+require 'test/unit'
+require "test/unit/runner/junitxml"
+require 'uri'
+require 'net/http'
+require './testutils'
+require 'json'
+
+BillingApi.configure do |config|
+  config.access_token = "fake"
+
+  # Changing host to mock server
+  config.host = '127.0.0.1:1080'
+  config.scheme = 'http'
+  config.server_index = nil
+end
+
+class TC_BillingApi < Test::Unit::TestCase
+
+  def verify_called_once(expectation)
+    # Result retrieved from server's verification
+    # Verifying expectation matched exactly once.
+    verify_result = TestUtils.verify_expectation_matched_times(expectation, 1)
+
+    # Asserts a successful result.
+    # If not successful, will print the body of verify_result to debug.
+    assert_equal '202', "#{verify_result.code}", "API should have been hit exactly once. MockServer returned #{verify_result.code}.\n #{verify_result.body}"
+  end
+
+  def setup
+    TestUtils.reset_mockserver
+  end
+
+  def teardown
+    TestUtils.reset_expectations
+  end
+
+  def test_get_account_billing_configuration_me
+    request, response = TestUtils.generate_payloads_from('billingapi/billing_configurations/account_billing_configurations_me_get')
+    expectation = TestUtils.setup_expectation(request, response, 1)
+
+    api_instance = BillingApi::BillingConfigurationsApi.new
+
+    result = api_instance.account_billing_configuration_me_get
+
+    assert_equal response[:body], result.to_hash.compact
+
+    self.verify_called_once expectation
+  end
+
+  def test_get_product_availability
+    request, response = TestUtils.generate_payloads_from('billingapi/products/product_availability_get')
+    expectation = TestUtils.setup_expectation(request, response, 1)
+
+    api_instance = BillingApi::ProductsApi.new
+    opts = {
+      product_category: ['SERVER'],
```

Comment:
The utils method extracts a single list item as its own string, so instead of sending a list of 1 item, it sends only the 1 item. The issue here is that `product_availability_get` is expecting certain opt properties to be lists, so I have to generate it manually. 

I couldn't add a second value to the list in the payload because in the case of `product_category` and `solution`, there is only 1 valid value.

Reply:
Fair enough. Although we could technically update the utils, might not be worth it for edge cases like this considering we only have one API afaik with list-query-params.